code/game/objects/effects/decals/cleanable.dm

```diff
@@ -17,11 +17,77 @@
 
 	var/on_wall = 0 //Wall on which this decal is placed on
 
-/obj/effect/decal/cleanable/New()
-	if(random_icon_states && length(src.random_icon_states) > 0)
+	var/persistence_type = SS_CLEANABLE
+	var/age = 1 //For map persistence. +1 per round that this item has survived. After a certain amount, it will not carry on to the next round anymore.
+	var/persistent_type_replacement //If defined, the persistent item generated from this will be of this type rather than our own.
+
+/obj/effect/decal/cleanable/New(var/loc, var/age, var/icon_state, var/color, var/dir, var/pixel_x, var/pixel_y)
+	if(age)
+		setPersistenceAge(age)
+	if(icon_state)
+		src.icon_state = icon_state
+	else if(random_icon_states && length(src.random_icon_states) > 0)
 		src.icon_state = pick(src.random_icon_states)
+	if(color)
+		src.color = color
+	if(dir)
+		src.dir = dir
+	if(pixel_x)
+		src.pixel_x = pixel_x
+	if(pixel_y)
+		src.pixel_y = pixel_y
+
+	if(ticker)
+		initialize()
+
+	..(loc)
+
+	blood_list += src
```

Comment:
man... you need to clean up this stuff on Destroy()

Reply:
nevermind, there's two `/obj/effect/decal/cleanable/Destroy()`
fuck me

Reply:
I have said it before and I'll say it again: multiple declarations of a proc in separate files are a mistake which should be purged